partner_contact_address_default/views/res_partner_views.xml

```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="view_partner_form" model="ir.ui.view">
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="base.view_partner_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='child_ids']" position="before">
+                <group>
+                    <group>
+                        <field
+                            name="partner_delivery_id"
+                            domain="[('id', 'child_of', commercial_partner_id), ('type', '=', 'delivery')]"
```

Comment:
This part of the domain `('id', 'child_of', commercial_partner_id)` gives a warning in case `commercial_partner_id` is `False`.

The warning is:

```log
2024-09-24 12:56:05,649 55 WARNING test_database odoo.osv.expression: Unexpected domain [('id', 'child_of', False)], interpreted as False
```
and it comes from here:

https://github.com/odoo/odoo/blob/17.0/odoo/osv/expression.py#L871

Reply:
I think that https://github.com/OCA/partner-contact/pull/1838 fixes the warning above.